src/Controller/Hello.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\ControllerTrait;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class Hello
+{
+    use ControllerTrait;
+
+    /**
+     * @Route("/", name="homepage")
+     */
+    public function __invoke(Request $request): Response
```

Comment:
@yceruto in HttpKernel, the controller is not the class. It is always the callable.

A controller cannot have actions. A single class can contain multiple controllers. Controllers are **always** methods in Symfony (unless you go back to symfony 1 where the term `controller` had a different meaning).
Btw, when using Silex, it is very common to avoid using classes entirely and use closures instead (which cannot be done in the fullstack framework as it does not play well with caching)